pkg/controller/atlasproject/project_settings.go

```diff
@@ -0,0 +1,73 @@
+package atlasproject
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func ensureProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) (result workflow.Result) {
+	if result = syncProjectSettings(ctx, projectID, project); !result.IsOk() {
+		ctx.SetConditionFromResult(status.ProjectSettingsReadyType, result)
+		return result
+	}
+
+	if areProjectSettingsEmpty(project.Spec.Settings) {
+		ctx.UnsetCondition(status.ProjectSettingsReadyType)
+		return workflow.OK()
+	}
+
+	ctx.SetConditionTrue(status.ProjectSettingsReadyType)
+	return workflow.OK()
+}
+
+func syncProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) workflow.Result {
+	spec := project.Spec.Settings
+
+	atlas, err := fetchSettings(ctx, projectID)
+	if err != nil {
+		return workflow.Terminate(workflow.ProjectSettingsReady, err.Error())
+	}
+
+	if !areSettingsInSync(atlas, spec) {
+		if err := patchSettings(ctx, projectID, spec); err != nil {
```

Comment:
I see here the we take k8s spec as source of truth. Considering the import feature, shouldn't we merge k8s spec into existing atlas settings instead of overwrite it? (this would require a upper-cased info in the docs to avoid misunderstandings).
Ignore my comment if this behavior was previously discussed and defined as it is.

Reply:
Export feature will just take the current settings via GET request and create a spec with these Project Settings so we don't need to do any merging here. And also the spec should be taken as source of truth in most cases.